marlowe-dashboard-client/src/MainFrame/State.purs

```diff
@@ -235,12 +259,14 @@ handleAction (PickupAction (Pickup.PickupWallet walletDetails)) = do
       -- subscribe to the wallet to get updates about balance changes
       subscribeToWallet wallet
       liftEffect $ setItem walletDetailsLocalStorageKey $ encodeJSON updatedWalletDetails
-    -- TODO: show errors to the user
+    Failure err -> networkErrorToast err
     _ -> pure unit
 
 handleAction (PickupAction Pickup.GenerateNewWallet) = do
   remoteDataWalletInfo <- createWallet
   assign (_newWalletDetails <<< _remoteDataWalletInfo) remoteDataWalletInfo
+  let
+    networkErrorToast = \_ -> addToast $ connectivityErrorToast "Couldn't generate wallet"
```

Comment:
I feel this (and line 234 above) should be factored out somehow, but I'm not immediately sure how. Just a note for the future...